### PR TITLE
New Bootstrap-UI

### PR DESCRIPTION
**Pull request by tamaskan**
Updated JQuery
Included Standard Roboto-Font
Custom Date-Format via http://momentjs.com/
Coin-Icon-Font via https://labs.allienworks.net/icons/cryptocoins/
Added Total Account-Value and Value per Coin
Different collapsible Sections and Filters
Progressbars and visual Min-Coin-Warning
Increased max. Update-Interval to 3600

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [N/A] **I have read CONTRIBUTING.md**
- [N/A] **I fully understand [Github Flow.](https://guides.github.com/introduction/flow/)**
- [X] **My code adheres to the [code style of this project.](https://poloniexlendingbot.readthedocs.io/en/latest/contributing.html)**
- [N/A] **I have updated the documentation in /docs if I have changed the config, arguments, logic in how the bot works, or anything that understandably needs a documentation change.**
- [N/A] **I have updated the config file accordingly if my change requires a new configuration setting or changes an existing one.**
- [N/A] **I have tested the bot with no issues for 24 continuous hours. If issues were experienced, they have been patched and tested again.**

### CODE REVIEW

**Comment by rnevet** [comment]
General comment, use spaces, keep things correctly aligned.

**Comment by laxdog** [comment]
Do we need all the binary files? It's generaly a bad idea to commit binaries to git w/o annex or something similar.

**Comment by tamaskan** [comment]
I haven't found a hosted version of the cryptocoin-font, but it could be reduced to a single .woff2-file

**Comment by rnevet** [comment]
'Show the amount of earned Coins since first UI-Start (Stored in
localstorage)'

Not needed, we are going to have complete lending history in this branch:
https://github.com/BitBotFactory/poloniexlendingbot/pull/349

Please correct tabs as it's really hard to review the code this way.

**Comment by rnevet** [comment]
Would it make sense to use https://rawgit.com/ for the fonts files, so they are loaded faster?

**Comment by rnevet** [comment]
While I find displaying the opened loans interesting, displaying the cancel loan events is useless as this happens on every bot cycle and doesn't actually bring any information.

**Comment by rnevet** [comment]
image

**Comment by rnevet** [comment]
The filtering buttons are ignored when a reload happens. :)

**Comment by rnevet** [comment]
I like the overall design, however I think many of the features should be discussed. 

In general I would prefer to have first design change and then separate PRs for features... this will also make it easier to review and approve.

**Comment by tamaskan** [comment]
The creators of this bot are free to use only the non-controversial parts and open issues for discussions :-)

File: www/lendingbot.js [hunk 1]
```diff
@@ -24,15 +24,80 @@ var Satoshi = new BTCDisplayUnit("Satoshi", 100000000);
...
 var btcDisplayUnitsModes = [BTC, mBTC, Bits, Satoshi];
 
+// Date Format
+    var dateformat = localStorage.getItem('dateformat') || 'DD. M. YYYY H:mm:ss ';
+    
+//Account value
+    var totalvalue = 0;
+    
+//Min Coins based on https://github.com/BitBotFactory/poloniexlendingbot/issues/347
+    var items = [['BTC','BTS','CLAM','DASH','DOGE','ETH','FCT','LTC','MAID','STR','XMR','XRP'],[0.00019124,10,10,0,100,0,100,0,10,100,0,100]];
```
Comment (Evanito) [inline www/lendingbot.js]: Bitcoin min is actually 0.01, in the issue he was demonstrating the different values of the minimums.

Comment (tamaskan) [inline www/lendingbot.js]: Is there a list of these minimums ?

**Comment by kingo55** [comment]
The design changes look really nice!

Hiding a bunch of stuff under accordions makes this design less practical though. I like to be able to view all the stats upon a glance. If I have to tap to open accordions, it takes time and effort. Arguably not much, but enough that it makes the old design more useful at this stage. 

It gets really annoying when the bot refreshes the page and you lose all the accordion states! :(

**Comment by tamaskan** [comment]
I can change that behavior, thanks for the feedback :-)

**Comment by tamaskan** [comment]
https://github.com/dutu/poloLender has a clean interface too, if you want a different approach

**Comment by rnevet** [comment]
How about using prettyFloat instead of printFloat ? :)

**Comment by tamaskan** [comment]
No problem, the function-description is quite vague ^^

**Comment by rnevet** [comment]
Well it prints a pretty float with a specified accuracy.
i.e. accuracy 2:
1.02332341 -> 1.02**34**
4556.1233 -> 456.**12**
1.00 -> 1
1.10 -> 1.1
etc...

**Comment by LeaderGL** [comment]
Hi,
with your branch i always receive the "DataTables warning: table id={id} - Requested unknown parameter '{parameter}' for row {row-index}, column{column-index}`" error and my "Opened Loan" tabs is empty even if i have opened loan.

File: www/lendingbot.js [hunk 2]
```diff
@@ -24,25 +24,296 @@ var Satoshi = new BTCDisplayUnit("Satoshi", 100000000);
...
+       mydatatable = $('#openloans').dataTable();
+    }
+    else {
+       mydatatable = $('#openloans').dataTable({responsive: true,createdRow: function( row, data, dataIndex ) {
+          $( row ).find('tr').addClass($( row ).find('td:eq(2) span').attr('data-type'));
+        }});
+    }
+
+    mydatatable.fnClearTable();
+    mydatatable.fnAddData(dataSet);
```
Comment (LeaderGL) [inline www/lendingbot.js]: what if "dataSet" is empty?
we can incur in "DataTables warning: table id={id} - Requested unknown parameter '{parameter}' for row {row-index}, column{column-index}`" error

File: www/lendingbot.js [hunk 3]
```diff
@@ -24,25 +24,296 @@ var Satoshi = new BTCDisplayUnit("Satoshi", 100000000);
...
+    }
+    else {
+       logtable = $('#logtable').dataTable({responsive: true,createdRow: function( row, data, dataIndex ) {
+          $( row ).find('td:eq(1) a').attr('data-order',$( row ).find('td:eq(1) a').attr('data-date'));
+          $( row ).addClass($( row ).find('td:eq(2) span').attr('data-type'));        
+        }});
+    }
+
+    logtable.fnClearTable();
+    logtable.fnAddData(logdataSet);
```
Comment (LeaderGL) [inline www/lendingbot.js]: what if "logdataSet" is empty?
we can incur in "DataTables warning: table id={id} - Requested unknown parameter '{parameter}' for row {row-index}, column{column-index}`" error

**Comment by tamaskan** [comment]
interesting, never experienced this problem. how does your botlog.json look like ?

**Comment by LeaderGL** [comment]
at fail time my botlog is like that (i've removed my coin amount):
`{"last_status": "Lent: [0.xxxxxxxx BTC @ 0.1462%] ", "last_update": "2017-06-22 22:44:27", "log": ["2017-06-22 22:44:08 Using custom mindailyrate 0.100% for BTC", "2017-06-22 22:44:14 Not lending BTC due to rate below 0.1561%"], "outputCurrency": {"currency": "BTC", "highestBid": "1"}, "raw_data": {"BTC": {"averageLendingRate": "0.1461733644848254994269884326", "lentSum": "0.xxxxxxxx", "maxToLend": "0.xxxxxxxx", "totalCoins": "0.xxxxxxxx"}}}`

**Comment by tamaskan** [comment]
If there is no active Loan, i would run into the Problem again, so i just added a Check :-)
It would be nice if emilam could push the py-Changes, so i can adapt my Code.

**Comment by rnevet** [comment]
btw, I made a JSFiddle for prettyFloat
https://jsfiddle.net/mut72zcp/

**Comment by rnevet** [comment]
@tamaskan how about making this an alternate theme based on:
https://github.com/BitBotFactory/poloniexlendingbot/commit/86c3521f65df2ef41bd41a62c5b34c45c644cd81
?

**Comment by mchl18** [comment]
some friends and i will work on an alternative frontend soon, but probably on my fork. It really needs an overhaul

**Comment by rnevet** [comment]
@mchl18 have you seen the option to create alternate themes?

**Comment by tamaskan** [comment]
@rnevet sure, that is not a problem.
@mchl18 awesome :-)

**Comment by mchl18** [comment]
also wanna try and write a config interface...wip

**Comment by mchl18** [comment]
@rnevet no i have not. I just fiddled with some css overlay

**Comment by tamaskan** [comment]
thought about a configuration-interface, but the main problem is to keep it up to date

**Comment by tamaskan** [comment]
great :-) do you need help ?

**Comment by mchl18** [comment]
I havent had time to set it up, wanna meet with some friends to discuss, if I run into questions i will let you know

**Comment by mchl18** [comment]
vue.js seems tempting but idk if it can handle the whole parsing and all. also i wanted to create tiles for the currencies instead of tables to make it more customisable and responsive

**Comment by tamaskan** [comment]
never worked with that, but it should be able to handle that.

**Comment by mchl18** [comment]
yeah it looks really slim and easy to set up, much better than angular and simpler than react, but also only js at its core

**Comment by utdrmac** [comment]
@tamaskan What's the status on this? There are some other PRs that want to make UI changes too but this is holding that up and will make the merge of this PR harder. Are you pausing this effort for the moment?

**Comment by tamaskan** [comment]
yes, i am still working on this. the bot supports multiple page-templates, so i see no reason why another developer can't create a PR with his Idea.

**Comment by rnevet** [comment]
@utdrmac as @tamaskan a recent change to the bot was allowing multiple html templates (IMO might be harder to maintain but it is in). 
https://github.com/BitBotFactory/poloniexlendingbot/commit/86c3521f65df2ef41bd41a62c5b34c45c644cd81

**Comment by gincarbone** [comment]
great job

**Comment by gincarbone** [comment]
if someone want to contribute to a project of a trading bot that operates on GDAX (coinbase) take a look at this: https://github.com/gincarbone/roy

**Comment by bkuhl** [comment]
Seems like a while since this has had a contribution.  Is this still a work in progress?  Can a screenshot be provided?

**Comment by bkuhl** [comment]
Still no updates here.... hrmmmm... what do we do with this PR?

**Comment by utdrmac** [comment]
Is this PR dead?

**Comment by SpiryMedia** [comment]
I do think it deserves a neat UI. And the initial PR had a very strong idea in that direction.

**Comment by tamaskan** [comment]
i switched to coinlend.org, but everyone is free to use the code and create something beautiful

**Pull request closed without merging**